### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by mrocklin** [ContinuumIO/dask#4]
When I run the large computation at the start of [this blog post](http://matthewrocklin.com/blog/work/2015/01/06/Towards-OOC-Scheduling/) my process uses up a bunch of memory.  All `dask.threaded` specific diagnostics say that this shouldn't be the case (we seem to be releasing memory appropriately).

I don't really know what's going on here.  Using `guppy` on the python process doesn't report the data.  Checking the size of the data in the `state['cache']` during execution looks great.  Running heavy concurrent HDF5 reads doesn't result in a leak.  Turning off HDF5 writing (more often known to cause leaks) doesn't solve the problem.  Flushing the file at each write also doesn't do anything.

So in short, I don't know what's going on.  Fixing this is probably the largest looming issue.

**Comment by mrocklin** [comment]
It could be that fixing the leak is hard (requires in depth understanding of existing scheduler code) but that just identifying the leak is easy.  I would be very glad if someone identified the leak and told me about it.

**Comment by mrocklin** [comment]
Fixed by @nevermindewe in #6
